app/services/user_authenticator.rb

```diff
@@ -0,0 +1,77 @@
+class UserAuthenticator
```

Comment:
**Code quality - Refactoring**

Introduce the `UserAuthenticator` service object to remove utility logic from the `UserModel` and the `UserController`.